concent_api/core/tests/test_payments_sci_callback.py

```diff
@@ -0,0 +1,192 @@
+            gasprice=10 ** 6,
+            startgas=80000,
+            value=10,
+            to=b'7917bc33eea648809c28',
+            v=self.v,
+            r=self.r,
+            s=self.s,
+            data=b'3078666333323333396130303030303030303030303030303030303030303030303032393662363963383738613734393865663463343531363436353231336564663834666334623330303030303030303030303030303030303030303030303030333431336437363161356332633362656130373531373064333239363566636161386261303533633030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303237313030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303562343336643666',
+        )
+
+    def _create_unsigned_transaction(self):  # pylint: disable=no-self-use
+        return Transaction(
+            nonce=99,
+            gasprice=10 ** 6,
+            startgas=80000,
+            value=10,
+            to=b'7917bc33eea648809c28',
+            data=b'3078666333323333396130303030303030303030303030303030303030303030303032393662363963383738613734393865663463343531363436353231336564663834666334623330303030303030303030303030303030303030303030303030333431336437363161356332633362656130373531373064333239363566636161386261303533633030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303237313030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303562343336643666',
+        )
+
+    def test_that_sci_callback_should_sign_transaction(self):
+        transaction = self._create_unsigned_transaction()
+
+        with mock.patch('core.payments.sci_callback.socket.socket.connect'):
+            with mock.patch('core.payments.sci_callback.send_over_stream'):
+                with mock.patch('core.payments.sci_callback.unescape_stream'):
+                    with mock.patch(
+                        'middleman_protocol.message.AbstractFrame.deserialize',
+                        return_value=GolemMessageFrame(payload=self._create_signed_transaction(), request_id=99),
+                    ):
+                        signed_transaction = sci_callback(transaction)
+
+        self.assertEqual(signed_transaction.v, self.v)
+        self.assertEqual(signed_transaction.r, self.r)
+        self.assertEqual(signed_transaction.s, self.s)
+
+    def test_that_sci_callback_should_raise_exception_on_timeout(self):
+        transaction = self._create_unsigned_transaction()
+
+        with mock.patch('core.payments.sci_callback.socket.socket.connect'):
+            with mock.patch('core.payments.sci_callback.send_over_stream', side_effect=socket.timeout):
+                with self.assertRaises(SCICallbackTimeoutError):
+                    sci_callback(transaction)
+
+    def test_that_sci_callback_should_raise_exception_on_receiving_incorrectly_signed_middleman_protocol_message(self):
```

Comment:
Does this exception `MiddlemanProtocolError` always mean that MiddleMan protocol message is incorrectly signed?

Reply:
No, I have updated test name. :+1: